libsolidity/analysis/TypeChecker.cpp

```diff
@@ -2073,6 +2074,20 @@ bool TypeChecker::visit(MemberAccess const& _memberAccess)
 					errorMsg += " Use \"address(" + varName + ")." + memberName + "\" to access this address member.";
 					break;
 				}
+		}
+		else if (exprType->category() == Type::Category::Address)
```

Comment:
```suggestion
		else if (auto addressType = dynamic_cast<AddressType const*>(exprType.get()))
```
?
That checks the category and casts in one go... The same could be done elsewhere around here as well, e.g. for the ``Type::Category::Function`` case above.